magic_the_gathering/game_state.py

```diff
@@ -68,18 +76,23 @@ def set_decks(self, decks: List[List[Card]]):
         self.decks = decks
 
     def cast_card(self, player_index, hand_card_index):
+        # TODO: Should we implement a play_card method? Is there a difference between casting and playing a card?
         card = self.hands[player_index].pop(hand_card_index)
-        self.stack.append(StackItem(card=card, owner_player_index=player_index))
+        self.stack.append(card)
 
     def resolve_top_of_the_stack(self):
-        stack_item = self.stack.pop(-1)
-        board_item = BoardItem(card=stack_item.card, state=CardPosition.TAPPED)
-        self.boards[stack_item.owner_player_index].append(board_item)
+        # TODO: Currently, this was implemented with casting creatures or lands in mind. It should be generalized.
+        card = self.stack.pop(-1)
+        card.is_tapped = True
+        self.boards[card.owner_player_index].append(card)
 
-    def untap_all_permanents(self, player_index):
-        for board_item in self.boards[player_index]:
-            board_item.state = CardPosition.UNTAPPED
+    def untap_all_permanents(self, player_index: int):
+        for card in self.boards[player_index]:
```

Comment:
I'd go with permanent instead of card
Also, we should probably have a terminology for the card meaning the blueprint from which to instantiate card objects, or an instance of it
`permanent` could be an instance of a `permanent_card`, `object` could be an instance of a `card`

Reply:
I renamed the variable :slightly_smiling_face: However, it is possible that in other places the naming is wrong (I have not checked) :sweat_smile:

Reply:
I agree with having a terminology, let's discuss in the future how to implement this properly :wink: